src/message.rs

```diff
@@ -18,6 +19,16 @@ pub fn process(
     )
   })?;
 
+  if let Some(destination_path) = parameters.destination_path {
+    fs::write(&destination_path, &result).map_err(|e| {
+      let error_result = job_result
+        .clone()
+        .with_status(JobStatus::Error)
+        .with_message(&e.to_string());
+      MessageError::ProcessingError(error_result)
+    })?;
```

Comment:
replace using helper: [MessageError::from](https://docs.rs/mcai_worker_sdk/0.11.11/mcai_worker_sdk/enum.MessageError.html#impl)
it can look like:
```rust
    fs::write(&destination_path, &result).map_err(|io_error| {
      MessageError::from(io_error, job_result.clone())
    })?;
```